File: java/main/WeightedRandomException.java
package weightedrandom.main;

public class WeightedRandomException extends RuntimeException {
    
    public WeightedRandomException() {}
    
    public WeightedRandomException(String message) {
        super(message);
    }
    
    public WeightedRandomException(Throwable cause) {
        super(cause);
    }
    
    public WeightedRandomException(String message, Throwable cause) {
        super(message, cause);
    }
}

File: C#/WeightedRandomException.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace WeightedRandom
{
    [Serializable]
    public class WeightedRandomException : Exception 
    {
        public WeightedRandomException()
        {
        }

        public WeightedRandomException(string message) 
            : base(message)
        {
        }

        public WeightedRandomException(string message, Exception inner) 
            : base(message, inner)
        {
        }
    }
}